extensions/s3/src/main/java/io/deephaven/extensions/s3/S3ChannelContext.java

```diff
@@ -331,24 +333,34 @@ private String requestStr() {
 
         final class Sub implements Subscriber<ByteBuffer> {
             private final CompletableFuture<ByteBuffer> localProducer;
-            private ByteBuffer bufferView;
+            // Access to this view must be guarded by bufferReference.acquire
+            private final ByteBuffer bufferView;
             private Subscription subscription;
 
             Sub() {
                 localProducer = producerFuture;
-                if (!bufferReference.acquire()) {
+                final ByteBuffer get;
+                if ((get = bufferReference.acquireAndGet()) == null) {
```

Comment:
I think you should use the `acquireIfAvailable` pattern here for consistency, and just do the `get()` below as I had it.